lib/ansible/playbook/base.py

```diff
@@ -567,7 +567,14 @@ def post_validate_attribute(self, name: str, *, templar: TemplateEngine):
             return Sentinel
 
         # FIXME: compare types, not strings
-        if not attribute.always_post_validate and self.__class__.__name__ not in ('Task', 'Handler', 'PlayContext', 'IncludeRole', 'TaskInclude'):
+        if not attribute.always_post_validate and self.__class__.__name__ not in (
+            "Task",
+            "Handler",
+            "PlayContext",
+            "IncludeRole",
+            "TaskInclude",
+            "HandlerTaskInclude",
+        ):
```

Comment:
This isn't required for this PR, but I wonder if it might be more beneficial to add a class attraction or something that defines whether the container (class) shoudl be post validated. Rather than supplying a list of the class names. Just a thought.